src/sentry/replays/usecases/query/__init__.py

```diff
@@ -43,6 +43,47 @@
 from sentry.replays.usecases.query.fields import ComputedField, TagField
 from sentry.utils.snuba import raw_snql_query
 
+VIEWED_BY_ME_KEY_ALIASES = ["viewed_by_me", "seen_by_me"]
+NULL_VIEWED_BY_ID_VALUE = 0  # default value in clickhouse
+
+
+def handle_viewed_by_me_filters(
+    search_filters: Sequence[SearchFilter | str | ParenExpression], request_user_id: int | None
+) -> Sequence[SearchFilter | str | ParenExpression]:
+    """Translate "viewed_by_me" as it's not a valid Snuba field, but a convenience alias for the frontend"""
+    new_filters = []
+    for search_filter in search_filters:
+        if (
+            not isinstance(search_filter, SearchFilter)
+            or search_filter.key.name not in VIEWED_BY_ME_KEY_ALIASES
+        ):
+            new_filters.append(search_filter)
+            continue
+
+        if request_user_id is None:
+            raise NotAuthenticated("No request user provided for viewed_by_me filter")
```

Comment:
Ok, got it. My thought process was this might only apply to this filter, but I guess it's worth validating for the whole get method!

Reply:
This breaks type-checking though, since calls to the query fxs with request_user=None (e.g. from the delete_replays script) pass through handle_viewed_by_me. How should we handle this?